src/ray/raylet/task.h

```diff
@@ -73,6 +73,10 @@ class Task {
   /// false otherwise.
   bool DependsOn(const ObjectID &object_id) const;
 
+  /// Update the dynamic/mutable information for this task.
+  /// \param task Task structure with updated dynamic information.
+  void Update(const Task &task);
```

Comment:
I think I would probably give this a more descriptive name, since we want to be careful when using this method. Maybe something like `CopyTaskExecutionSpec`.